source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -25,1144 +25,3357 @@
-    #endif /* ( ipconfigHAS_DEBUG_PRINTF == 0 ) */
-#endif /* ifdef ipconfigHAS_DEBUG_PRINTF */
 
-#ifndef FreeRTOS_debug_printf
-    #define FreeRTOS_debug_printf( MSG )    do {} while( ipFALSE_BOOL )
-    /* MISRA Ref 20.5.1 [Use of undef] */
-    /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-2051 */
-    /* coverity[misra_c_2012_rule_20_5_violation] */
-    #undef ipconfigHAS_DEBUG_PRINTF
-    #define ipconfigHAS_DEBUG_PRINTF    0
+#ifndef ipconfigRA_SEARCH_COUNT
+    #define ipconfigRA_SEARCH_COUNT    3
 #endif
 
-/*
- * FreeRTOS general logging routine (proposal)
- * Used in some utility functions such as FreeRTOS_netstat() and FreeRTOS_PrintARPCache()
- *
- *     extern void my_printf( const char * pcFormat, ... );
- *     #define FreeRTOS_printf( MSG )           my_printf MSG
- *
- * The FreeRTOS_printf() must be thread-safe but does not have to be interrupt-safe
- */
-#ifdef ipconfigHAS_PRINTF
-    #if ( ipconfigHAS_PRINTF == 0 )
-        #ifdef FreeRTOS_printf
-            #error Do not define FreeRTOS_print if ipconfigHAS_PRINTF is set to 0
-        #endif /* ifdef FreeRTOS_debug_printf */
-    #endif /* ( ipconfigHAS_PRINTF == 0 ) */
-#endif /* ifdef ipconfigHAS_PRINTF */
+#if ( ipconfigRA_SEARCH_COUNT < 0 )
+    #error ipconfigRA_SEARCH_COUNT must be at least 0
+#endif
 
-#ifndef FreeRTOS_printf
-    #define FreeRTOS_printf( MSG )    do {} while( ipFALSE_BOOL )
-    /* MISRA Ref 20.5.1 [Use of undef] */
-    /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-2051 */
-    /* coverity[misra_c_2012_rule_20_5_violation] */
-    #undef ipconfigHAS_PRINTF
-    #define ipconfigHAS_PRINTF    0
+#if ( ipconfigRA_SEARCH_COUNT > UINT_FAST8_MAX )
+    #error ipconfigRA_SEARCH_COUNT overflows a UBaseType_t
 #endif
 
+/*-------------------------------------------------------------------*/
+
 /*
- * In cases where a lot of logging is produced, FreeRTOS_flush_logging( )
- * will be called to give the logging module a chance to flush the data.
+ * ipconfigRA_SEARCH_TIME_OUT_MSEC
+ *
+ * Type: TickType_t
+ * Unit: milliseconds
+ * Minimum: 0
+ * Maximum: portMAX_DELAY * portTICK_PERIOD_MS
+ *
+ * Sets the timeout to wait for a response to a router
+ * solicitation message.
  */
-#ifndef FreeRTOS_flush_logging
-    #define FreeRTOS_flush_logging()    do {} while( ipFALSE_BOOL )
-#endif
 
-/* Malloc functions.  Within most applications of FreeRTOS, the couple
- * pvPortMalloc()/vPortFree() will be used.
- * If there are different types of RAM, the user may decide to use a different
- * memory allocator for different purposes:
- * MallocLarge is used to allocate large TCP buffers (for Rx/Tx)
- * MallocSocket is used to allocate the space for the sockets
- */
-#ifndef pvPortMallocLarge
-    #define pvPortMallocLarge( x )    pvPortMalloc( x )
+#ifndef ipconfigRA_SEARCH_TIME_OUT_MSEC
+    #define ipconfigRA_SEARCH_TIME_OUT_MSEC    10000
```

Comment:
```suggestion
    #define ipconfigRA_SEARCH_TIME_OUT_MSEC    ( 10000 )
```

Reply:
Should they all be like this?

Reply:
Yes. We'd suggest to use brackets to wrap the macros like guidelines in [FreeRTOS Coding Standard, Testing and Style Guide](https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html).
Below is the example under Style Guide.
`#define A_DEFINITION	( 1 )`